src/components/Search/SearchView.tsx

```diff
@@ -1,17 +1,37 @@
-import React from 'react';
+import * as React from 'react';
 
 import Box from '@mui/material/Box';
 import Container from '@mui/material/Container';
 import Grid from '@mui/material/Grid';
 import Typography from '@mui/material/Typography';
 
-import zap from '../../theme/img/zap-10.svg';
+import Zap from '../../theme/img/zap-10.svg';
 import SearchDropdown from './SearchDropdown';
 import SearchInput from './SearchInput';
 
 export default function SearchView() {
+  const [focused, setFocused] = React.useState(false);
+
+  const handleParentClick = (e: React.MouseEvent) => {
+    if ((e.target as HTMLElement).id === 'search-revision-input') {
+      setFocused(true);
+    } else {
+      setFocused(false);
+    }
+  };
+
+  const handleFocus = (e: React.FocusEvent) => {
+    e.stopPropagation();
+    setFocused(true);
+  };
+
+  const handleChildClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    setFocused(true);
+  };
+
```

Comment:
these functions are replacing `handleClickOutsideInput` and using bubbling instead of hard-coded IDs or class names

`stopPropagation` prevents event handlers of parents from running when a child is clicked/focused

Reply:
I added this here instead of a helper file because I'm using `useState` to set the value of `focused`. I may be able to move this out, but I was running into issue with passing `setFocused` as a parameter due to not setting the type correctly